### PR TITLE
Allows humans with a robotic voicebox installed, MMIs, and posibrains to use silicon emotes, and simple/basic bots to beep

### PR DESCRIPTION
**Pull request by 00-Steven**
### EDITED, see spoiler for previous about section.

This pr adds the `TRAIT_SILICON_EMOTES_ALLOWED`, which is now used to determine what can perform silicons emotes instead of typechecks.
We do this by adding a `trait_required` var to emotes, which if set is checked for in `can_run_emote(...)` and means the mob must have that trait to use the emote.

We then adds this trait to all of the mobs previously capable of using silicon emotes and the beep emote, being:
All silicons, basic bots, simple bots, orbies, and brain mobs.
In addition, we add the trait to robotic voicebox users.

Finally, it makes beeping a silicon emote.

What this changes is that robotic voicebox users can beep and use other silicon emotes, basic and simple bots can actually beep, sentient orbies could use other silicon emotes, and brains can beep and use silicon emotes.
Previously brains where on the allowed types list for beeping, but living emotes also blocked brains generally, so they couldn't _actually_ beep.

<details>
  <summary>Previous About Section</summary>
  
All this really does is add a new trait, `TRAIT_SILICON_EMOTES_ALLOWED`, which when given to a human allows them to use silicon emotes.
It does this by adding the `/mob/living/carbon/human` type to the allowed types for these emotes, but then adding an additional check to `can_run_emote(...)` for these emotes to block it for humans if they don't have the trait.
We add it to the typecache because the base checks for `can_run_emote(...)` would block if not for that, but we _do_ want the other checks it runs.
  
</details>

## Why It's Good For The Game

I just think it'd be fun, really. Like, _I'd_ want to beep at people when I'm going out of my way to augment the shit out of my character.

It's an item primarily accessible by roboticists and transhumanists, both of which are encouraged/required to interact with silicons, so I think this'd lead to fun interactions for those that do commit to getting the silicon-speech tongue replacement.

Finally, I mean, it's a funky synthesizer that makes you sound like a silicon. Why shouldn't it let you synthesize these sounds too? Hell, even its attack verbs are beeping and booping already.

### EDITED, additional justification:

It's weird for basic/simple bots to be able to use all other silicon emotes except beeping.

MMI'd brains/posibrains being able to beep looked intentional, just broken. This fixes that.
I feel we might as well just allow them to use other silicon emotes while we're at it.

## Changelog
:cl:
add: Having a robotic voicebox installed now lets you use silicon emotes.
balance: MMI'd brains/posibrains can now use silicon emotes.
balance: Simple and basic bots can actually beep.
fix: MMI'd brains/posibrains are no longer both allowed AND blocked from beeping, now just being allowed to beep.
/:cl:

### CODE REVIEW

**Comment by Ghommie** [comment]
The robotic voicebox should also give you the silicon tts filter, but maybe that's just my opinion.

**Comment by 00-Steven** [comment]
I believe it already has a filter that's at least _very similar_, buuuut I can't really find where the silicon voice filter is determined to confirm silicons also use the exact same one.
Notably it's the same one as vendors use.

**Comment by Ghommie** [comment]
Oh, It already does, my bad. Also the actual silicon filter is inside the tts py script, the TTS_FILTER_SILICON define is just a string.

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -720,9 +720,14 @@
 	message = "beeps."
 	message_param = "beeps at %t."
 	sound = 'sound/machines/twobeep.ogg'
-	mob_type_allowed_typecache = list(/mob/living/brain, /mob/living/silicon, /mob/living/basic/orbie)
+	mob_type_allowed_typecache = list(/mob/living/carbon/human, /mob/living/brain, /mob/living/silicon, /mob/living/basic/orbie)
```
Comment (MrMelbert) [inline code/modules/mob/living/emote.dm:723]: You know if we wanted to we could just nuke the typecache entirely and give the trait to silicons/brains/orbs in `Initialize` as an `INNATE_TRAIT`.

Food for thought

Comment (00-Steven) [reply code/modules/mob/living/emote.dm:723]: Not a bad thought... I'll think about it

**Comment by zombiegums** [comment]
Very cute idea, def see fun robo/borg interactions popping up with this.

**Comment by 00-Steven** [comment]
drafting to look into melbert's suggestion

**Comment by 00-Steven** [comment]
So while looking into Melbert's suggestion, I realized, why in the everliving fuck is *beep specifically a living emote instead of a silicon, when it's only ever used by things that're arguably synthetic? And as such has its own typechecks, meaning that for some reason simple and basic bots can't beep.
Similarly, orbies (virtual pet thingy) can only beep and perform none of the other silicon emotes.
MMI'd brains ALSO can only beep... except they can't beep, because living emotes _also_ blacklist brains from beeping.

Soooooo I'm just gonna move beep to be a silicon emote, move all silicon emotes to checking the trait, and give it to silicons, robotic voicebox users, simple and basic bots, brains, and orbies, and call it a day.

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (00-Steven) [reply code/modules/mob/living/emote.dm]: thought about it-

**Comment by 00-Steven** [comment]
Aaaaaand PR body and name updated.

**Comment by Dawnseer** [comment]
Very nice

**Pull request merged**